components/exercises/chapter-4-oscar/AList/listToArray.test.tsx

```diff
@@ -0,0 +1,57 @@
+import { listToArray } from './listToArray';
+
+describe('listToArray', () => {
+  it('should convert a linked list to an array', () => {
+    const listNum = {
+      value: 1,
+      rest: {
+        value: 2,
+        rest: {
+          value: 3,
+          rest: null,
+        },
+      },
+    };
+
+    const listString = {
+      value: 'a',
+      rest: {
+        value: 'b',
+        rest: {
+          value: 'c',
+          rest: null,
+        },
+      },
+    };
+
+    const expectedLongerListString = {
+      value: 'a',
+      rest: {
+        value: 'b',
+        rest: {
+          value: 'c',
+          rest: {
+            value: 'd',
+            rest: {
+              value: 'f',
+              rest: null,
+            },
+          },
+        },
+      },
+    };
+
+    const expectedNumArray = [1, 2, 3];
+    const expectedStringArray = ['a', 'b', 'c'];
+    const expectedLongerStringArray = ['a', 'b', 'c', 'd', 'f'];
+    expect(listToArray(listNum)).toEqual(expectedNumArray);
+    expect(listToArray(listString)).toEqual(expectedStringArray);
+    expect(listToArray(expectedLongerListString)).toEqual(
+      expectedLongerStringArray
+    );
+  });
```

Comment:
This is actually 3 tests, can we split it?
```
it('should convert a linked list with numbers to an array', () => {

it('should convert a linked list with strings to an array', () => {

it('should convert a longer linked list with strings to an array', () => {
```

We generally want to make tests atomic, so if there's some change that breaks the code fewer tests will fail and it will be easier to debug

Reply:
refactored, thank you